### PR TITLE
feat(component_tag_release.groovy): add <component>-tag-release jobs

### LINKED ISSUE #158: [Meta] <component>-tag-release job
As part of the initiative to release a single component at a time (independent of Workflow as a whole) we require a `<component>-tag-release` job for each, which should:
1. watch for git push events to a refspec of `+refs/tags/*:refs/remotes/origin/tags/*` and branch `*/tags/*` (in essence, a tag push event)
2. determine if branch from above is indeed the latest tag, if so, proceed; if no, exit.
3. based on commit at HEAD of branch, find the correlating `git-<commit:0:7>` image tag (ref https://github.com/deis/jenkins-jobs/issues/155) in the 'deis' org of quay and use this for running/vetting in e2e tests
4. assuming e2e passes, retag image from above with 'pretty'/semver release tag (based on actual git tag pushed) in the quay.io and dockerhub 'deis' orgs

TODO for an iterative approach
- [x] create job covering steps 1. and 2. (https://github.com/deis/jenkins-jobs/pull/168)
- [x] promote prod-candidate images on master merge https://github.com/deis/jenkins-jobs/issues/155
- [x] once the e2e strategy is finalized, add step 3. and 4. to job (https://github.com/deis/jenkins-jobs/issues/169)

ref https://github.com/deis/workflow/pull/410

### LINKED ISSUE #169: [component release] add e2e and final retag jobs to component release pipeline
https://github.com/deis/jenkins-jobs/pull/168 introduces the first stage of a component release pipeline as described in https://github.com/deis/jenkins-jobs/issues/158.

After this is done, we will need to finish steps 3. and 4. in the https://github.com/deis/jenkins-jobs/issues/158 mentioned above.

### PR DESCRIPTION
**Pull request by vdice**
ref https://github.com/deis/jenkins-jobs/issues/158
closes https://github.com/deis/jenkins-jobs/issues/169
closes https://github.com/deis/jenkins-jobs/issues/158

TODO
- [x] remove [these lines](https://github.com/vdice/jenkins-jobs/blob/9d3df91d871188c5b9b6390edef57852e6de8b38/jobs/component_tag_release.groovy#L5-L6) used for testing.
- [x] get bash logic under test

### CODE REVIEW

**Comment by vdice** [comment]
First successful release pipeline in https://ci.deis.io/job/router-tag-release/9/console consisting of 1. locate release candidate -> 2. run e2e against candidate -> 3. retag candidate w/ semver tag

File: bash/scripts/locate_release_candidate.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  envPropsFilepath="${BATS_TEST_DIRNAME}/tmp/env.properties"
+  if [ -n "${JENKINS_HOME}" ]; then
+    envPropsFilepath="${WORKSPACE}/env.properties"
+  fi
+
+  tag=''
+  if [ -n "${TAG}" ]; then
+    echo "TAG set to '${TAG}', attempting release of this tag..."
+    tag="${TAG}"
+  else
```
Comment (bacongobbler) [inline bash/scripts/locate_release_candidate.sh]: could we change this logic to use an `elif` instead to lower the cyclomatic complexity?

File: bash/scripts/locate_release_candidate.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if [ -n "${GIT_BRANCH}" ]; then
+      echo "GIT_BRANCH set to '${GIT_BRANCH}', attempting release of this tag..."
+      tag="${GIT_BRANCH#origin/tags/}"
+    else
+      echo "GIT_BRANCH not set, cannot determine tag to release; exiting."
```
Comment (bacongobbler) [inline bash/scripts/locate_release_candidate.sh]: in regards to that same comment, can we change this to "GIT_BRANCH or TAG not set, cannot determine tag to release; exiting."?

File: bash/scripts/locate_release_candidate.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      exit 1
+    fi
+
+    # make sure tag matches latest; else, exit
```
Comment (bacongobbler) [inline bash/scripts/locate_release_candidate.sh:19]: Would there ever be a case where we'd want to release an older tag like a patch release for an older release?

File: bash/scripts/locate_release_candidate.sh (hunk 1 above)
Comment (vdice) [inline bash/scripts/locate_release_candidate.sh]: good call; changed.

File: bash/scripts/locate_release_candidate.sh (hunk 2 above)
Comment (vdice) [inline bash/scripts/locate_release_candidate.sh]: changed as well with the `elif` switch.

File: bash/scripts/locate_release_candidate.sh (hunk 3 above)
Comment (vdice) [inline bash/scripts/locate_release_candidate.sh:19]: I think indeed there could be @bacongobbler -- currently it is possible to do so by kicking off pipeline manually with a certain `TAG` value.

Comment (bacongobbler) [inline bash/scripts/locate_release_candidate.sh:19]: sgtm, just wanted to make sure we can enable that. :)

**Pull request merged**